breadbox/breadbox/api/types.py

```diff
@@ -580,6 +595,36 @@ def list_dimension_types_endpoint(db: SessionWithUser = Depends(get_db_with_user
     return [_dim_type_to_response(x) for x in dim_types]
 
 
+@router.get(
+    "/dimensions/{name}/identifiers",
+    operation_id="get_dimension_type_identifiers",
+    response_model=List[DimensionIdentifiers],
+)
+def get_dimension_type_identifiers(
+    name: str,
+    data_type: Annotated[Union[str, None], Query()] = None,
+    show_only_dimensions_in_datasets: Annotated[Union[bool, None], Query()] = None,
+    db: SessionWithUser = Depends(get_db_with_user),
+):
+    dim_type = type_crud.get_dimension_type(db, name)
+    if dim_type is None:
+        raise HTTPException(404, f"Dimension type {name} not found")
+
+    if show_only_dimensions_in_datasets is True:
+        dimension_ids_and_labels = metadata_service.get_dimension_type_identifiers(
+            db, dim_type, data_type, show_only_dimensions_in_datasets=True
+        )
+    else:
+        dimension_ids_and_labels = metadata_service.get_dimension_type_identifiers(
+            db, dim_type, data_type
+        )
```

Comment:
Oh I see now, I hadn't noticed that you were setting `Union[bool, None]` as the typing for that `show_only_dimensions_in_datasets` parameter.

But I am curious why that parameter needs to be `Union[bool, None]` instead of just a `bool`? Couldn't you update the typing to be something like:
```
show_only_dimensions_in_datasets: Annotated[bool, Query()] = False,
```
I think that would make it more clear what the default behavior is. Also, I just tried it out locally and it seems to work fine at least with the breadbox client (screenshot below), but I'd be curious if you've run into different issues?

 
<img width="507" alt="image" src="https://github.com/user-attachments/assets/d41fc80a-3faa-4dd7-8e96-6316e8eefeaa">

Reply:
Ah maybe I misinterpreted the FastAPI docs. I think you're right. It makes sense that query params should already be optional without needing to explicitly pass the default value `=None`